src/main/java/seedu/address/logic/util/ModuleClassModificationUtil.java

```diff
@@ -0,0 +1,119 @@
+        assert targetModuleClass != null;
+        assert lessonToEdit != null;
+        assert editedLesson != null;
+        assert targetModuleClass.hasLesson(lessonToEdit);
+
+        if (!lessonToEdit.isSameLesson(editedLesson) && targetModuleClass.hasLesson(editedLesson)) {
+            throw new CommandException(MESSAGE_DUPLICATE_LESSON);
+        }
+
+        Name moduleClassName = targetModuleClass.getName();
+        Set<UUID> studentsIds = new HashSet<>(targetModuleClass.getStudentUuids());
+        List<Lesson> listOfLessons = targetModuleClass.getLessons();
+        List<Lesson> editedListOfLessons = new ArrayList<>();
+        for (Lesson lesson : listOfLessons) {
+            if (lesson.equals(lessonToEdit)) {
+                editedListOfLessons.add(editedLesson);
+            } else {
+                editedListOfLessons.add(lesson);
+            }
+        }
+        assert listOfLessons.size() == editedListOfLessons.size();
+        return new ModuleClass(moduleClassName, studentsIds, editedListOfLessons);
+    }
+
+    /**
+     * Adds all lessons in the target module class to the new module class.
+     * The {@code lessonToDelete} is deleted from the new module class.
+     *
+     * @throws CommandException if the {@code lessonToDelete} already exists.
+     */
+    public static ModuleClass createModifiedModuleClassWithDeletedLesson(
+            ModuleClass targetModuleClass, Lesson lessonToDelete) {
+        assert targetModuleClass != null;
+        assert lessonToDelete != null;
+        assert targetModuleClass.hasLesson(lessonToDelete);
+
+        Name moduleClassName = targetModuleClass.getName();
+        Set<UUID> studentsIds = new HashSet<>(targetModuleClass.getStudentUuids());
+        List<Lesson> listOfLessons = targetModuleClass.getLessons();
+        List<Lesson> editedListOfLessons = new ArrayList<>();
+        for (Lesson lesson : listOfLessons) {
+            if (!lesson.equals(lessonToDelete)) {
+                editedListOfLessons.add(lesson);
+            }
+        }
+        assert listOfLessons.size() - 1 == editedListOfLessons.size();
+        return new ModuleClass(moduleClassName, studentsIds, editedListOfLessons);
+    }
+
+    /**
+     * Updates {@code lessons} in {@code moduleClass} and creates a modified {@code moduleClass} object.
+     */
+    public static ModuleClass createModifiedModuleClassWithModifiedLesson(
+            ModuleClass targetModuleClass, Index lessonToEditIndex, Lesson lessonToUpdate) {
+        assert targetModuleClass != null;
+        assert lessonToEditIndex != null;
+        assert lessonToUpdate != null;
+
+        Name moduleClassName = targetModuleClass.getName();
+        Set<UUID> studentsIds = targetModuleClass.getStudentUuids();
+        List<Lesson> lessons = new ArrayList<>(targetModuleClass.getLessons());
+        lessons.set(lessonToEditIndex.getZeroBased(), lessonToUpdate);
+
+        return new ModuleClass(moduleClassName, studentsIds, lessons);
+    }
```

Comment:
Can I check how does this defer in purpose from `createModifiedModuleClassWithEditedLesson`?

Reply:
```createModifiedModuleClassWithModifiedLesson``` is used in AttendanceCommands